diff --git a/src/filemgr.hxx b/src/filemgr.hxx
index 0603b7d72de8..7773a321a66e 100644
--- a/src/filemgr.hxx
+++ b/src/filemgr.hxx
@@ -72,25 +72,6 @@
 #ifndef FILEMGR_HXX_
 #define FILEMGR_HXX_
 
-#include "hunzip.hxx"
-#include <cstdio>
-#include <string>
-#include <fstream>
+#include "mozHunspellRLBoxSandbox.h"
 
-class FileMgr {
- protected:
-  std::ifstream fin;
-  Hunzip* hin;
-  char in[BUFSIZE + 50];  // input buffer
-  int fail(const char* err, const char* par);
-  int linenum;
-
- public:
-  FileMgr(const char* filename, const char* key = nullptr);
-  FileMgr(const FileMgr&) = delete;
-  FileMgr& operator=(const FileMgr&) = delete;
-  ~FileMgr();
-  bool getline(std::string&);
-  int getlinenum();
-};
 #endif
