https://chromium-review.googlesource.com/c/libyuv/libyuv/+/6772565

diff --git a/libyuv.gyp b/libyuv.gyp
index d4973b0..e853ba3 100644
--- a/libyuv.gyp
+++ b/libyuv.gyp
@@ -26,30 +26,15 @@
     'clang%': 0,
     # Link-Time Optimizations.
     'use_lto%': 0,
-    # Enable LASX on LoongArch by default.
-    "loong_lasx%": 1,
-    # Enable LSX on LoongArch by default. Has no effect if loong_lasx is
-    # enabled because LASX implies LSX according to the architecture specs.
-    "loong_lsx%": 1,
     'mips_msa%': 0,  # Default to msa off.
     'build_neon': 0,
-    "build_lasx": 0,
-    "build_lsx": 0,
     'build_msa': 0,
-
     'conditions': [
        ['(target_arch == "armv7" or target_arch == "armv7s" or \
        (target_arch == "arm" and arm_version >= 7) or target_arch == "arm64")\
        and (arm_neon == 1 or arm_neon_optional == 1)', {
          'build_neon': 1,
        }],
-       ['(target_arch == "loong64") and (loong_lasx == 1)', {
-         "build_lasx": 1,
-         "build_lsx": 1,  # LASX implies LSX.
-       }],
-       ['(target_arch == "loong64") and (loong_lsx == 1)', {
-         "build_lsx": 1,
-       }],
        ['(target_arch == "mipsel" or target_arch == "mips64el")\
        and (mips_msa == 1)',
        {
@@ -75,16 +60,6 @@
             '-Wno-unused-parameter',
          ],
         }],
-        ["build_lasx != 0", {
-          "cflags": ["-mlasx"],
-        }, {  # build_lasx == 0
-          "cflags": ["-mno-lasx"],
-        }],
-        ["build_lsx != 0", {
-          "cflags": ["-mlsx"],
-        }, {  # build_lsx == 0
-          "cflags": ["-mno-lsx"],
-        }],
         ['build_neon != 0', {
           'defines': [
             'LIBYUV_NEON',

