diff --git a/src/common/runtime/helper/test_worker.ts b/src/common/runtime/helper/test_worker.ts
index 7b7922a88f..9e10b666da 100644
--- a/src/common/runtime/helper/test_worker.ts
+++ b/src/common/runtime/helper/test_worker.ts
@@ -200,7 +200,9 @@
     const selfPathDir = selfPath.substring(0, selfPath.lastIndexOf('/'));
     // Construct the path to the worker file, then use URL to resolve the `../` components.
     const serviceWorkerURL = new URL(
-      `${selfPathDir}/../../../${suite}/webworker/${fileName}.as_worker.js`
+      // NOTE: This is a Mozilla-specific hack! Please follow-up at
+      // <https://bugzilla.mozilla.org/asdf> for removal.
+      `${location.origin}/_mozilla/webgpu/${suite}/webworker/${fileName}.as_worker.js`
     ).toString();
 
     // Ensure the correct service worker is registered.
