diff --git a/src/llama-sampling.cpp b/src/llama-sampling.cpp
index dd1d9eee52..41081da1ba 100644
--- a/src/llama-sampling.cpp
+++ b/src/llama-sampling.cpp
@@ -305,9 +305,10 @@
 }
 
 static uint32_t get_rng_seed(uint32_t seed) {
+
     if (seed == LLAMA_DEFAULT_SEED) {
         // use system clock if std::random_device is not a true RNG
-        static bool is_rd_prng = std::random_device().entropy() == 0;
+        static bool is_rd_prng = true;//std::random_device().entropy() == 0;
         if (is_rd_prng) {
             return (uint32_t) std::chrono::system_clock::now().time_since_epoch().count();
         }
