diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
--- a/expat/lib/xmlparse.c
+++ b/expat/lib/xmlparse.c
@@ -115,6 +115,8 @@
 #include "expat.h"
 #include "siphash.h"
 
+/* BEGIN MOZILLA CHANGE (we already set a salt through XML_SetHashSalt16Bytes) */
+#if 0
 #if defined(HAVE_ARC4RANDOM)
 #  include "random_arc4random.h"
 #endif /* defined(HAVE_ARC4RANDOM) */
@@ -138,6 +140,8 @@
 #if defined(_WIN32)
 #  include "random_rand_s.h"
 #endif /* defined(_WIN32) */
+#endif
+/* END MOZILLA CHANGE */
 
 #if ! defined(HAVE_GETRANDOM) && ! defined(HAVE_SYSCALL_GETRANDOM)             \
     && ! defined(HAVE_ARC4RANDOM_BUF) && ! defined(HAVE_ARC4RANDOM)            \
@@ -1036,6 +1036,8 @@
        ASCII_s,     ASCII_p,     ASCII_a,      ASCII_c,      ASCII_e,
        '\0'};
 
+/* BEGIN MOZILLA CHANGE (we already set a salt through XML_SetHashSalt16Bytes) */
+#if 0
 #if ! defined(HAVE_ARC4RANDOM_BUF) && ! defined(HAVE_ARC4RANDOM)
 
 static unsigned long
@@ -1062,9 +1064,14 @@
 }
 
 #endif /* ! defined(HAVE_ARC4RANDOM_BUF) && ! defined(HAVE_ARC4RANDOM) */
+#endif
+/* END MOZILLA CHANGE */
 
 static struct sipkey
 ENTROPY_DEBUG(const char *label, struct sipkey entropy_128) {
+/* BEGIN MOZILLA CHANGE (we already set a salt through XML_SetHashSalt16Bytes) */
+  (void)label;
+#if 0
   if (getDebugLevel("EXPAT_ENTROPY_DEBUG", 0) >= 1u) {
     fprintf(stderr,
             "expat: Entropy: %s --> [0x" EXPAT_FMT_LLX(
@@ -1072,11 +1079,16 @@
             label, (unsigned long long)entropy_128.k[0],
             (unsigned long long)entropy_128.k[1]);
   }
+#endif
+/* END MOZILLA CHANGE */
   return entropy_128;
 }
 
 static struct sipkey
 generate_hash_secret_salt(void) {
+/* BEGIN MOZILLA CHANGE (we already set a salt through XML_SetHashSalt16Bytes) */
+  abort();
+#if 0
   struct sipkey entropy;
 
   /* "Failproof" high quality providers: */
@@ -1125,6 +1137,8 @@
     return ENTROPY_DEBUG("fallback(8)", entropy);
   }
 #endif
+#endif
+/* END MOZILLA CHANGE */
 }
 
 static enum XML_Error
